common/dao-api/src/main/java/org/thingsboard/server/dao/firmware/FirmwareService.java

```diff
@@ -18,17 +18,25 @@
 import com.google.common.util.concurrent.ListenableFuture;
 import org.thingsboard.server.common.data.Firmware;
 import org.thingsboard.server.common.data.FirmwareInfo;
+import org.thingsboard.server.common.data.exception.ThingsboardException;
+import org.thingsboard.server.common.data.firmware.ChecksumAlgorithm;
 import org.thingsboard.server.common.data.id.FirmwareId;
 import org.thingsboard.server.common.data.id.TenantId;
 import org.thingsboard.server.common.data.page.PageData;
 import org.thingsboard.server.common.data.page.PageLink;
 
+import java.nio.ByteBuffer;
+
 public interface FirmwareService {
 
     FirmwareInfo saveFirmwareInfo(FirmwareInfo firmwareInfo);
 
     Firmware saveFirmware(Firmware firmware);
 
+    PageData<String> getSupportedChecksumAlgorithms(PageLink pageLink);
```

Comment:
Why this is pageData and pageLink? More then 100 algorithms supported? UI will use paginated requests?